src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -731,131 +725,150 @@ internal void DeleteTransientNodes()
+                    {
+                        newNode.X = offset; // Adjust X position
+                        newNode.Y = PortViewModel.NodeViewModel.NodeModel.Y; // Adjust Y position
+                        workspaceModel.AddAndRegisterNode(newNode);
+                        createdNodes[nodeItem.Id] = newNode;
+                        createdClusterItems.Add(newNode);
+
+                        var newNodeViewModel = workspaceViewModel.Nodes.Last();
+                        newNodeViewModel.IsHidden = true; // Hide the node initially
+                    }
                 }
             }
 
-            if (createdNodes.Any())
+            // Create connections between nodes
+            foreach (var connection in clusterResultItem.Topology.Connections)
             {
-                targetNodeFromCluster = createdNodes[entryNodeId];
-            }
-
-            clusterConnections.ForEach(connection =>
-            {
-                // Connect the nodes    
-                var sourceNode = clusterMapping[connection.StartNode.NodeId].NodeModel;
-                var targetNode = clusterMapping[connection.EndNode.NodeId].NodeModel;
-                // The port index is 1- based (currently a hack and not expected from service)
-                var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.Index == connection.StartNode.PortIndex - 1);
-                var targetPort = targetNode.InPorts.FirstOrDefault(p => p.Index == connection.EndNode.PortIndex - 1);
-
-                if (targetPort != null && targetPort.Connectors.Count == 0)
+                if (createdNodes.TryGetValue(connection.StartNode.NodeId, out var sourceNode) &&
+                    createdNodes.TryGetValue(connection.EndNode.NodeId, out var targetNode))
                 {
-                    var connector = ConnectorModel.Make(sourceNode, targetNode, connection.StartNode.PortIndex - 1, connection.EndNode.PortIndex - 1);
+                    var sourcePortIndex = connection.StartNode.PortIndex - 1;
+                    var targetPortIndex = connection.EndNode.PortIndex - 1;
 
-                    newNodesAndWires.Add(connector);
-                }
+                    if (sourceNode.OutPorts.Count > sourcePortIndex && targetNode.InPorts.Count > targetPortIndex)
+                    {
+                        if (!targetNode.InPorts[targetPortIndex].Connectors.Any())
+                        {
+                            var newConnector = ConnectorModel.Make(sourceNode, targetNode, sourcePortIndex, targetPortIndex);
 
-            });
+                            if (newConnector != null)
+                            {
+                                newConnector.IsHidden = true; // Hide the connector initially
+                                createdClusterItems.Add(newConnector);
+                            }
+                        }
+                    }
+                }
+            }
 
             // Connect the cluster to the original node and port
```

Comment:
I feel we should prioritize this instead of the connection between the cluster nodes.
So we should try to do the connection between query node and the cluster node first ?
I feel like it looks more finished this way ? What do you think ? We can revisit later as well.